### PR TITLE
#448/Checking the autorun input should trigger run.

### LINKED ISSUE #448: Autorun won't run on first attempt
# 🐛 Bug Report

On fresh start, checking "autrun"  does not trigger a run.

## How to reproduce

Steps to reproduce the issue:

 0. Clear local storage
 1. Open the application in a browser
 2. Check "autorun" checkbox. Notice that no run is triggered.
 3. "Run" button is disabled - impossible to trigger manually. 

Application reaches an invalid state and it is not intuitive.

## 😯 Current Behavior

On fresh start, checking "autrun"  does not trigger a run.

## 🤔 Expected Behavior

On fresh start, checking "autrun"  triggers a run.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Browser          |
| Operating System |
|                  |

## Related

 -

### PR DESCRIPTION
**Pull request by abrie**
## Related issues and PRs

Fixes #448 

## Description

This changes the `useEffect` condition that runs the simulation: If `autorunSimulation` is changed to `true`, then the algorithm is run immediately and on subsequent parameter changes. 

## Impacted Areas in the application

Autorun behavior, URL location behavior.

### Autorun behavior
1. Open a fresh browser session.

2. Click the 'autorun' checkbox.
- The simulation should run.
- The 'run' button should be disabled.
3. Change a parameter in the Scenario card.
 - The simulation should run.
4. Reload the page.
 - The autorun checkbox should be checked (persisted).
 - The simulation should run.
 - The 'run' button should be disabled.
5. Un-check the autorun box.
 - The simulation should not run.
 - The 'run' button should be enabled.
6. Reload the page
 - The simulation should not run.
 - The 'run' button should be enabled.
 - The autorun checkbox should be unchecked.

(**E2E test needed**)

### CODE REVIEW

File: src/components/Main/Main.tsx [hunk 1]
```diff
@@ -114,6 +114,10 @@ function Main() {
   )
 
   useEffect(() => {
+    if (autorunSimulation) {
+        debouncedRun(allParams, scenarioState, severity)
+    }
+
```
Comment (abrie) [inline src/components/Main/Main.tsx:120]: Runs the simulation if autorun changed to true, regardless of whether parameters in URL location have changed.

File: src/components/Main/Main.tsx [hunk 2]
```diff
@@ -114,6 +114,10 @@ function Main() {
   )
 
   useEffect(() => {
+    if (autorunSimulation) {
+      debouncedRun(allParams, scenarioState, severity)
+    }
+
```
Comment (abrie) [inline src/components/Main/Main.tsx:120]: Runs the simulation if autorun changed to true, regardless of whether parameters in URL location have changed.

File: src/components/Main/Main.tsx [hunk 3]
```diff
@@ -125,7 +129,6 @@ function Main() {
 
       if (autorunSimulation) {
         updateBrowserURL(nextLocationSearch)
-        debouncedRun(allParams, scenarioState, severity)
       }
```
Comment (abrie) [inline src/components/Main/Main.tsx:132]: Update browser URL parameters if location search changed and autorunSimulation is true.

**Comment by ivan-aksamentov** [comment]
@Joluma Could you please take a look at how we can test this? Thanks!

**Pull request merged**